examples/platforms/utils/mac_frame.cpp

```diff
@@ -380,20 +380,24 @@ otError otMacFrameProcessTransmitSecurity(otRadioFrame *aFrame, otRadioContext *
 #if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
 void otMacFrameUpdateTimeIe(otRadioFrame *aFrame, uint64_t aRadioTime, otRadioContext *aRadioContext)
 {
-    if (aFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0)
-    {
-        uint8_t *timeIe = aFrame->mPsdu + aFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
-        uint64_t time   = aRadioTime + aFrame->mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset;
+    uint8_t *timeIe;
+    uint64_t time;
+
+    VerifyOrExit((aFrame->mInfo.mTxInfo.mIeInfo != nullptr) && (aFrame->mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0));
```

Comment:
If `mTimeIeOffset` is non-zero, should we require `aFrame->mInfo.mTxInfo.mIeInfo` to be also set?

Reply:
The `mTimeIeOffset` means `The Time IE offset from the start of PSDU.`. So when the time sync is working, the value `mTimeIeOffset` should not be 0. But we need to consider that when the program starts and the initial packets are sent, the default value of `mTimeIeOffset` is 0. The initial packets should not include the time sync IE. Besides, the code logic of [time sync](https://github.com/openthread/openthread/blob/main/src/core/mac/mac.cpp#L1077) shows that the `mTimeIeOffset` has a chance to be set to 0.

So I think it is necessary to check both conditions here.